server/Arcadia.Assistant.CSP/Vacations/VacationsSyncExecutor.cs

```diff
@@ -0,0 +1,185 @@
+﻿namespace Arcadia.Assistant.CSP.Vacations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Microsoft.EntityFrameworkCore;
+
+    using Arcadia.Assistant.Calendar.Abstractions;
+    using Arcadia.Assistant.CSP.Model;
+
+    public class VacationsSyncExecutor
+    {
+        private readonly Func<ArcadiaCspContext> contextFactory;
+
+        public VacationsSyncExecutor(Func<ArcadiaCspContext> contextFactory)
+        {
+            this.contextFactory = contextFactory;
+        }
+
+        public async Task<IEnumerable<Vacation>> UpsertVacation(CalendarEvent @event, VacationApprovals approvals, VacationsMatchInterval matchInterval)
+        {
+            var vacation = this.CreateVacationFromCalendarEvent(@event, approvals);
```

Comment:
please add some NLog's (debug level) with a count of existing vacations, new vacations, ids. Preferably in each case branch so we know what happens.